### PR TITLE
[Model] Add APPNP model

### PR DESCRIPTION
**Pull request by aymenwah**
## Description
This PR implement dgl version of APPNP model according to the paper of "Predict then Propagate: Graph Neural Networks meet Personalized PageRank " and its original code [https://github.com/klicperajo/ppnp](https://github.com/klicperajo/ppnp). .

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)

- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR

## Changes
This PR uses DGL's API to implement APPNP model.

### CODE REVIEW

**Comment by mufeili** [comment]
Thank you for your contribution! We will review it soon.

File: examples/pytorch/appnp/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+Dependencies
+------------
+- PyTorch 0.4.1+
+- requests
+
+``bash
+pip install torch requests
+``
+
+Codes
```
Comment (mufeili) [inline examples/pytorch/appnp/README.md]: I guess this should just be "Code"?

File: examples/pytorch/appnp/train.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    parser.add_argument("--gpu", type=int, default=-1,
+            help="gpu")
+    parser.add_argument("--lr", type=float, default=1e-2,
+            help="learning rate")
+    parser.add_argument("--n-epochs", type=int, default=200,
+            help="number of training epochs")
+    parser.add_argument("--hidden_sizes", type=int, nargs='+', default=[64],
+            help="hidden unit sizes for appnp")
+    parser.add_argument("--k", type=int, default=10,
+                        help="Number of propagation steps")
```
Comment (mufeili) [inline examples/pytorch/appnp/train.py]: The indent of this line is different from the rest arguments, maybe be consistent? This is also the case for line 132.

File: examples/pytorch/appnp/appnp.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    def forward(self, features):
+        # prediction step
+        h = features
+        if self.dropout:
+            h = self.dropout(h)
+        h = self.activation(self.layers[0](h))
+        for layer in self.layers[1:-1]:
+            h = self.activation(layer(h))
+        h = self.layers[-1](h)
```
Comment (mufeili) [inline examples/pytorch/appnp/appnp.py]: Seems the original PyTorch implementation also allows to perform a dropout here as in [L34](https://github.com/klicperajo/ppnp/blob/master/ppnp/pytorch/ppnp.py#L34)?

File: examples/pytorch/appnp/appnp.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        # propagation step
+        self.cached_h = h
+        for _ in range(self.K):
+            # normalization by square root of src degree
+            h = h * self.g.ndata['norm']
+            self.g.ndata['h'] = h
+            self.g.update_all(fn.copy_src(src='h', out='m'),
```
Comment (mufeili) [inline examples/pytorch/appnp/appnp.py]: As also noted by @BarclayII , need to comment that the dropout on adjacency matrices/edges is not performed during message passing.

File: examples/pytorch/appnp/appnp.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+class APPNP(nn.Module):
+    def __init__(self,
+                 g,
+                 in_feats,
+                 hiddens,
+                 n_classes,
+                 activation,
+                 dropout,
+                 alpha,
+                 k):
```
Comment (mufeili) [inline examples/pytorch/appnp/appnp.py]: Maybe also allow the choice of using bias in linear layers as in [the original implementation](https://github.com/klicperajo/ppnp/blob/master/ppnp/pytorch/ppnp.py#L11)?

Comment (BarclayII) [reply examples/pytorch/appnp/appnp.py]: We don't have to.  The difference of whether having a bias in a linear layer is pretty minor compared to the adjacency dropout (which is possible but not that trivial to implement)

Comment (mufeili) [reply examples/pytorch/appnp/appnp.py]: I agree, this is just a note.

File: examples/pytorch/appnp/appnp.py (hunk 3 above)
Comment (aymenwah) [reply examples/pytorch/appnp/appnp.py]: done.

File: examples/pytorch/appnp/appnp.py (hunk 5 above)
Comment (aymenwah) [reply examples/pytorch/appnp/appnp.py]: For now I'm going to add a comment.

File: examples/pytorch/appnp/appnp.py (hunk 4 above)
Comment (mufeili) [reply examples/pytorch/appnp/appnp.py]: @aymenwah Almost done. Could you please also add an in-line comment here about the dropout on adjacency matrix?

Comment (aymenwah) [reply examples/pytorch/appnp/appnp.py]: done.
